novawallet/Common/Services/Web3AlertService/Web3AlertsSyncService.swift

```diff
@@ -0,0 +1,174 @@
+                return nil
+            }
+            let remoteSaveOperation = self.remoteSaveOperation(settings: newSettings)
+            let localSaveOperation = self.localSaveOperation(settings: newSettings)
+
+            let mapOperation = ClosureOperation {
+                try remoteSaveOperation.extractNoCancellableResultData()
+                try localSaveOperation.extractNoCancellableResultData()
+            }
+
+            mapOperation.addDependency(remoteSaveOperation)
+            mapOperation.addDependency(localSaveOperation)
+
+            return .init(targetOperation: mapOperation, dependencies: [remoteSaveOperation, localSaveOperation])
+        }
+        wrapper.addDependency(operations: [newSettingsOperation])
+        return wrapper.insertingHead(operations: [newSettingsOperation])
+    }
+
+    override func stopSyncUp() {
+        syncCancellable.cancel()
+    }
+
+    private func remoteSaveOperation(settings: LocalPushSettings) -> BaseOperation<Void> {
+        let saveSettingsOperation: AsyncClosureOperation<Void> = AsyncClosureOperation(cancelationClosure: {}) { responseClosure in
+            let database = Firestore.firestore()
+            let documentRef = database.collection("users").document(settings.identifier)
+            let encoder = Firestore.Encoder()
+            encoder.dateEncodingStrategy = .iso8601
+            try documentRef.setData(from: RemotePushSettings(from: settings), merge: true, encoder: encoder) { error in
+                if let error = error {
+                    responseClosure(.failure(error))
+                } else {
+                    responseClosure(.success(()))
+                }
+            }
+        }
+
+        return saveSettingsOperation
+    }
+
+    private func localSaveOperation(settings: LocalPushSettings) -> BaseOperation<Void> {
+        repository.saveOperation({
+            [settings]
+        }, {
+            []
+        })
+    }
+
+    private func fetchLocalSettingsOperation() -> BaseOperation<LocalPushSettings?> {
+        guard let documentUUID = settingsManager.pushSettingsDocumentId else {
+            return .createWithResult(nil)
+        }
+        return repository.fetchOperation(
+            by: { documentUUID },
+            options: .init()
+        )
+    }
+}
+
+extension Web3AlertsSyncService: Web3AlertsSyncServiceProtocol {
+    func save(settings: LocalPushSettings) -> BaseOperation<Void> {
+        FirebaseHolder.shared.configureApp()
+        settingsManager.pushSettingsDocumentId = settings.identifier
+        return localSaveOperation(settings: settings)
+    }
+
+    func update(token: String) -> CompoundOperationWrapper<Void?> {
+        guard let documentUUID = settingsManager.pushSettingsDocumentId else {
+            return .createWithError(Web3AlertsSyncServiceError.documentNotFound)
+        }
+
+        FirebaseHolder.shared.configureApp()
```

Comment:
We have several places in file where we do this. Probably better to move to the service factory before service creation and assume it called before service created.